src/main/java/io/confluent/connect/jdbc/sink/DbStructure.java

```diff
@@ -46,6 +45,12 @@ public DbStructure(DatabaseDialect dbDialect) {
     this.tableDefns = new TableDefinitions(dbDialect);
   }
 
+  // For testing
+  public DbStructure(DatabaseDialect dbDialect, TableDefinitions tableDefns) {
```

Comment:
This is due to: 

```
final TableDefinition tableDefn = tableDefns.get(connection, tableId);


    final Set<SinkRecordField> missingFields = missingFields(
        fieldsMetadata.allFields.values(),
        tableDefn.columnNames()
    );

```
Because tableDefns is created using a mocked DatabaseDialect, when it creates a tableDefn using a mocked connection and tableId, the method columnNames() throws a NPE. If you had a suggestion to overcome this without creating the constructor, I'd definitely be willing to go with that path.

Reply:
We control everything we mock, including the `Connection` and `TableId`, right? Can we set up mocking on those objects to simulate the behavior we want here?

Not sure if this'd be cleaner than the current approach but it seems strange to add a package-private testing-only constructor when the only explicit dependency of the `DbStructure` class is a `DatabaseDialect`.

Reply:
@aakashnshah looks like this still needs to be addressed?

Reply:
I've removed the extra constructor, and used the existing mocked `DatabaseDialect` class by mocking its methods to get the desired results. Let me know what you think.

Reply:
💯